fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -1254,7 +1254,7 @@ default boolean deleteRecord(@Nonnull Tuple primaryKey) {
     /**
      * Delete all the data in the record store.
      * <p>
-     * Everything except the store header is cleared from the database.
+     * Everything except the store header and index state information is cleared from the database.
```

Comment:
- I agree that it might be okay to simply including WriteOnly states to make the semantic simple. But this should be documented somewhere.

- Since `deleteAllRecords` does not delete index states anymore, should there be a new interface to delete all index states? Or do we already have one? And maybe we also need separate interfaces for the two states.

Reply:
Okay, I attempted to be more explicit about what happens to `WRITE_ONLY` indexes in the doc.

I think the correct way (after calling `deleteAllRecords`) of also clearing out any index states is to use `rebuildAllIndexes` if one wants to continue using the record store and to use `deleteStore` (instead of `deleteAllRecords`) if one doesn't. I've updated the docs with that; I also found a bug (#597) where we weren't updating the local record store state, so I have updated the tests to exercise that function more.